File: payment-service/src/main/java/com/reski/payment/dto/OrderCreatedEvent.java
package com.reski.payment.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

@Data
public class OrderCreatedEvent {

    @JsonProperty("ID")
    private String id;

    @JsonProperty("CustomerName")
    private String customerName;

    @JsonProperty("ProductName")
    private String productName;

    @JsonProperty("Quantity")
    private Integer quantity;

    @JsonProperty("Amount")
    private Double amount;

    @JsonProperty("Status")
    private String status;

    @JsonProperty("CreatedAt")
    private String createdAt;

    @JsonProperty("UpdatedAt")
    private String updatedAt;
}

File: service-order/features/order/dto/order_created_event.go
package dto

type OrderCreatedEvent struct {
	EventID      string  `json:"EventID"`
	ID           string  `json:"ID"`
	CustomerName string  `json:"CustomerName"`
	ProductName  string  `json:"ProductName"`
	Quantity     int     `json:"Quantity"`
	Amount       float64 `json:"Amount"`
	Status       string  `json:"Status"`
	CreatedAt    string  `json:"CreatedAt"`
	UpdatedAt    string  `json:"UpdatedAt"`
}
